### PR TITLE
Blobstore sync bug fix

### PR DESCRIPTION
**Pull request by jiangzhileaf**
1.Bug fix for blob sync frequency with time unit error.
2.Bug fix for blob sync delete file, add catch NoSuchFileException.
3.Bug fix for blob sync update blob flie, add catch FileNotFoundException

### CODE REVIEW

File: storm-server/src/main/java/org/apache/storm/blobstore/LocalFsBlobStore.java [hunk 1]
```diff
@@ -197,7 +198,7 @@ public void run() {
                     throw new RuntimeException(e);
                 }
             }
-        }, 0, ObjectReader.getInt(conf.get(DaemonConfig.NIMBUS_CODE_SYNC_FREQ_SECS)));
+        }, 0, ObjectReader.getInt(conf.get(DaemonConfig.NIMBUS_CODE_SYNC_FREQ_SECS))*1000);
```
[the remark this answers is not in the document]
Comment (HeartSaVioR) [reply storm-server/src/main/java/org/apache/storm/blobstore/LocalFsBlobStore.java]: Nice catch!

File: storm-server/src/main/java/org/apache/storm/blobstore/BlobStoreUtils.java [hunk 2]
```diff
@@ -191,6 +192,8 @@ public static boolean downloadUpdatedBlob(Map<String, Object> conf, BlobStore bl
                     out.close();
                 }
                 isSuccess = true;
+            } catch(FileNotFoundException fnf) {
```
Comment (HeartSaVioR) [inline storm-server/src/main/java/org/apache/storm/blobstore/BlobStoreUtils.java]: I'd rather leave it as it is, unless we feel confident that it is OK to swallow here.

Comment (revans2) [reply storm-server/src/main/java/org/apache/storm/blobstore/BlobStoreUtils.java]: I agree with @HeartSaVioR here.

**Comment by HeartSaVioR** [comment]
@jiangzhileaf Any updates?

File: storm-server/src/main/java/org/apache/storm/blobstore/BlobStoreUtils.java (hunk 2 above)
Comment (arunmahadevan) [reply storm-server/src/main/java/org/apache/storm/blobstore/BlobStoreUtils.java]: Btw, theres a race condition where the topology gets killed while the blob download is going on and apparently it throws a FNF exception here when invoking out.close().

**Comment by arunmahadevan** [comment]
Seems to fix a few race conditions, may be we can take this forward.

**Comment by HeartSaVioR** [comment]
@arunmahadevan I'd be really happy if you can take forward with this. Looks like we lost @jiangzhileaf.

**Comment by arunmahadevan** [comment]
@HeartSaVioR created https://github.com/apache/storm/pull/2871 and linked to a JIRA. 

Maybe good to get this in 2.0 release.

**Comment by srdo** [comment]
Closing, see reasoning at https://lists.apache.org/thread.html/8f694615d3397a37c54175302ad35663483ed48f1ec51509f8a89318@%3Cdev.storm.apache.org%3E

**Pull request closed without merging**